### PR TITLE
Update trivy dependency and fix the code due to breaking changes

### PR DESCRIPTION
**Pull request by domust**

### CODE REVIEW

File: artifact.go [hunk 1]
```diff
@@ -338,10 +334,6 @@ func (a Artifact) inspectLayer(ctx context.Context, diffID string, disabled []an
 	opts := analyzer.AnalysisOptions{Offline: a.artifactOption.Offline}
 	result := analyzer.NewAnalysisResult()
 	limit := semaphore.NewWeighted(parallel)
```
Comment (damejeras) [inline artifact.go:336]: we have to make sure parrallel parameter will be 1 everywhere, otherwise there will be a lot OOM kills

Comment (domust) [reply artifact.go]: I've made it configurable via artifact options, so the caller gets to set whatever they want.

File: artifact.go [hunk 2]
```diff
@@ -249,11 +249,7 @@ func (a Artifact) inspect(ctx context.Context, missingImageKey string, layerKeys
 	blobInfo := make(chan types.BlobInfo)
 
 	errCh := make(chan error)
-	limit := semaphore.NewWeighted(parallel)
-	if a.artifactOption.Slow {
```
Comment (matas-cast) [inline artifact.go:253]: Slow mode was removed or why did you remove it?

**Review by matas-cast — APPROVED** [review]

File: artifact.go [hunk 3]
```diff
@@ -108,7 +108,7 @@ func NewArtifact(img types.Image, log logrus.FieldLogger, c CacheClient, opt art
 		log:            log,
 		image:          img,
 		cache:          c,
-		walker:         walker.NewLayerTar(opt.SkipFiles, opt.SkipDirs, opt.Slow),
+		walker:         walker.NewLayerTar(opt.SkipFiles, opt.SkipDirs),
```
Comment (anjmao) [inline artifact.go:111]: Add test for artifact and Inspect method similar as we have for full collect https://github.com/castai/kvisor/blob/main/cmd/agent/imagescan/collector/collector_test.go#L30

**Review by anjmao — APPROVED** [review]
